### PR TITLE
[dev] AB#25753 - [Person Record 2.0] [page] [md] - Notes: Need to use mobile notes drawer UX. (All Browsers) 

### PR DESCRIPTION
**Pull request by fantaJinMode**
PR contains:
- Ability to pass custom prop named `mobileMaxWidth`.  It will enable drawer to show stylings as mobile drawers for viewport less than `mobileMaxWidth`

eg: Person record notes drawer for viewport less than 1018px width, should show mobile styling for notes drawer(Default drawer for mobile styling is 767px)

Note: Right now, I am passing `mobileMaxWidth` props to `drawer`, `FiltersDrawer`, `FilterRails`. For future enhancement, I am thinking of passing only `mobileMaxWidth` props to `drawer`. It can pass it to its children (FiltersDrawer, FilterRails). For that, I was thinking of using `React.context`. Please provide any suggestions/ideas for it.

### CODE REVIEW

File: docs/package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "reacm-cm-ui-docs",
```
Comment (groberts314) [inline docs/package-lock.json]: We should fix this typo (in parent `package.json` file of course): `react-cm-ui-docs`
First time I've noticed that! :-)
Definitely a NAB for this PR ... it's been broken a while now.

File: src/templates/drawer.jsx [hunk 2]
```diff
@@ -254,6 +259,7 @@ class Drawer extends React.Component {
         const layeredOffset = 11;
         const scrollPosition = window.pageYOffset;
         const zIndex = 10002; // adding 2 accounts for the frist .drawer and .drawer-dimmers- z-indexes
+        const drawerWidth = mobileMaxWidth ? mobileMaxWidth + 1 : DEFAULT_DRAWER_WIDTH; // This implies if mobileMaxWidth is set and maxWidth not set
```
Comment (groberts314) [inline src/templates/drawer.jsx]: Curious ... why + 1?

File: src/templates/drawer.jsx [hunk 3]
```diff
@@ -416,7 +423,7 @@ class Drawer extends React.Component {
                             autoHide
                         >
                             <div className="drawer-container-inner">
-                                {children}
+                                    {children}
```
Comment (groberts314) [inline src/templates/drawer.jsx]: Looks like we added extra whitespace here?  Any reason?  Could we revert that?

File: src/templates/filtersDrawer.jsx [hunk 4]
```diff
@@ -238,12 +241,13 @@ class FiltersDrawer extends React.Component {
 
         return (
             <MediaQuery
-                maxWidth={767}
+                maxWidth={mobileMaxWidth ? mobileMaxWidth : 767}
```
Comment (groberts314) [inline src/templates/filtersDrawer.jsx]: Could we do `mobileMaxWidth || 767`?

File: src/templates/drawer.jsx (hunk 2 above)
Comment (fantaJinMode) [reply src/templates/drawer.jsx]: Like if, `mobileMaxWidth=1018` then drawer width would be around 1018. Previously, it was handled like this for 767.

File: src/templates/drawer.jsx (hunk 3 above)
Comment (fantaJinMode) [reply src/templates/drawer.jsx]: Thanks @groberts314 , for catching it. Fixed.

File: src/templates/filtersDrawer.jsx (hunk 4 above)
Comment (fantaJinMode) [reply src/templates/filtersDrawer.jsx]: Fixed

File: docs/package-lock.json (hunk 1 above)
Comment (fantaJinMode) [reply docs/package-lock.json]: Fixed

Comment (groberts314) [reply docs/package-lock.json]: I fixed it in `/docs/package.json` in b3a5970 =)

**Comment by morethanfire** [comment]
:shipit:

**Pull request merged**